### ISSUE TITLE
Script variables reseting from ByVal to ByRef when any value changed

### ISSUE THREAD

**Issue by LethDavidson** [chrismaltby/gb-studio#1218]
Howdy,

There's a circumstance that will make any edit to a script reset all script-specific variables inside of it when any value is changed. 

**To Reproduce**
EDIT: Below DID reproduce it, but then something changed and it's no longer this consistant. no idea

If multiple internal script Vars are set to ByValue and are being assigned the same number (0 in this case), anything WITH that same number gets reset. 
SO if, say, Vars A-C are assigned 0, but Var F is assigned "1", and they're all set to ByVal, if i change Var A to anything else (or change any event anywhere in the scipt) then B and C will get reset to ByRef, but VarF will remain ByVal.

**Expected behavior**
i expect it to _not_ do that. 
An example, i was making a test script and assigned multiple vars as 0 to be able to use them in a GBVM script, and needed them set to ByValue because if they were at ByRef, by default, they'd all be set to like 30 or something (no idea?)
 
 
Attached are three images of it happening, a youtube video of it happening, and my project. The problem seems to exist in any script, but the three scripts of "VarGBVMTextTest_3, and it's two other clearly named broken versions" are where the problem is active.

This is in 3.1, the main release. 

![2022-10-03 04_58_49-GB Studio - SampleProject3 03To3 1 (modified)](https://user-images.githubusercontent.com/18093091/193551064-904b6dcb-5754-49f6-942b-e78471e71d8a.png)
![2022-10-03 04_58_52-GB Studio - SampleProject3 03To3 1 (modified)](https://user-images.githubusercontent.com/18093091/193551066-21659731-96cf-4c3d-9781-dcc1970e632c.png)
![2022-10-03 04_58_57-GB Studio - SampleProject3 03To3 1 (modified)](https://user-images.githubusercontent.com/18093091/193551068-140339bf-c46e-4870-a148-64df99e4b1d7.png)

https://www.youtube.com/watch?v=5KmiYTNn4gQ

[MySampleEdits.zip](https://github.com/chrismaltby/gb-studio/files/9696560/MySampleEdits.zip)

**Comment by RichardULZ** [comment]
Confirming, this is only an editor side bug, where the By Ref or By Value drop down on Variable A/B/C ect, are reset if the pass through variables are changed. 
Nothing to do with in game or another bug, just updating the order resets By Ref/By Var settings. (the video helps a lot, couldn't understand what you meant)

I can sort of see how that might be happening, and not exactly sure on a quick fix, or if it's intentional?...
When using Set variable to value, it sounds like it must be a reference, if you tried to set a number you passed through to a number in the script, that sounds pointless? 

It may not be able to modify a variable because it's only a value? Unless you expect it to generate a temporary variable for this script context, which i'm not sure on either (most variable pass through modify the outside variable directly i think).

**Comment by LethDavidson** [comment]
Yeah, i have a feel that in some capacity this is supposed to be happening, and it's partically my lack of care with what i was using and how they forced it into a situation where i was relying on something techincally broken for the thing to work. 
now granted, i KEPT having the same issue even when being more careful and it was always extremely unclear what exact what causing it or what it thinks i'm doing wrong.
I ended up writing a version that didn't run into those issues, but that was largely becuase i leaned far more on globals instead of locals cuz the script locals would just keep doing it.

**Comment by chrismaltby** [comment]
Hi @LethDavidson I think I've figured out the bug causing this issue there was a problem in the code that parsed variables in text and expression values causing it to believe the argument was always passByReference.

It was a two character change to fix in the end:
https://github.com/chrismaltby/gb-studio/commit/6aff4b17efffabd0c1012217e37d755306cb5b8b

Hopefully this was the same issue you were having here
